docs/src/pages/components/text-fields/text-fields.md

```diff
@@ -285,7 +285,8 @@ In order for the text field to be accessible, **the input should be linked to th
 </div>
 ```
 
-- If you are using the `TextField` component, you just have to provide a unique `id`.
+- If you are using the `TextField` component, you just have to provide a unique `id` unless you're using the `TextField` only client side.
+  Until the UI is hydrated `TextField` without an explicit `id` will not have associated labels.
```

Comment:
From what I understand in https://github.com/reactwg/react-18/discussions/111, doesn't the `useId` work on the server? I mean even though I do not specify `id` to `<TextField />`, the id will be generated by `useId` and that should work in both server & client?

Reply:
Only if you have React 18 installed. This sentence describes behavior in React 17 which is the only stable major we support. Though I'm hesitant to add this exception here since it incentivizes using React 18 when you shouldn't do that yet (in production). 

The `useMediaQuery` also doesn't have a note about improved behavior in React 18.

But it's clear that we should track changes to documentation we want to make once React 18 is out. We may need a guide for the styling solution when using streaming rendering anyway (see https://github.com/reactwg/react-18/discussions/113#discussioncomment-1672153).

Reply:
Added an item to https://github.com/mui-org/material-ui/issues/29844. Thanks for pointing this out 👍🏻

Reply:
Got it, this is nice. 🙏